server/trace.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var errors = require('../lib/errors');
+var safeParse = require('../lib/util').safeParse;
+var toCamelCase = require('../lib/util').toCamelCase;
+var traceCore = require('../lib/trace/core');
+
+module.exports = {
+    addTrace: {
+        endpoint: '/trace/add',
+        handler: createAddTrace,
+    },
+    removeTrace: {
+        endpoint: '/trace/remove',
+        handler: createRemoveTrace,
+    },
+};
+
+function createAddTrace(ringpop) {
+    return function addTrace(optsStr, _, hostInfo, cb) {
```

Comment:
Hrm. Leftovers. I agree w/ _ and _2 -- I miss languages that allow the duplicate _, _, convention. I will give them names, prefixed w/ _ (which I know as another convention for indicating ignored args; can also do _ign1 and _ign2, which is less weird than _ and _2).